packages/fluentui/react-northstar/src/components/Video/Video.tsx

```diff
@@ -24,77 +28,93 @@ export interface VideoProps extends UIComponentProps {
 
   /** Video source URL. */
   src?: string;
-}
-
-export const videoClassName = 'ui-video';
-
-class Video extends UIComponent<WithAsProp<VideoProps>> {
-  static create: ShorthandFactory<VideoProps>;
-
-  static deprecated_className = videoClassName;
-
-  static displayName = 'Video';
-
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      children: false,
-      content: false,
-    }),
-    autoPlay: PropTypes.bool,
-    controls: PropTypes.bool,
-    loop: PropTypes.bool,
-    muted: PropTypes.bool,
-    poster: PropTypes.string,
-    src: PropTypes.string,
-  };
-
-  static defaultProps = {
-    as: 'video',
-    controls: true,
-    autoPlay: false,
-    muted: false,
-  };
 
-  videoRef = React.createRef<HTMLVideoElement>();
+  accessibility?: Accessibility;
 
-  componentDidMount() {
-    this.setVideoAttributes();
-  }
+  variables?: ComponentVariablesInput;
+}
 
-  componentDidUpdate() {
-    this.setVideoAttributes();
-  }
+export const videoClassName = 'ui-video';
 
-  setVideoAttributes = () => {
+export const Video: React.FC<WithAsProp<VideoProps>> & FluentComponentStaticProps<VideoProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Video.displayName, context.telemetry);
+  setStart();
+  const { controls, autoPlay, loop, poster, src, muted, variables } = props;
+  const ElementType = getElementType(props);
+  const videoRef = React.createRef<HTMLVideoElement>();
+  const unhandledProps = useUnhandledProps(Video.handledProps, props);
+  const getA11yProps = useAccessibility(props.accessibility, {
+    debugName: Video.displayName,
+  });
+
+  const setVideoAttributes = () => {
     // React doesn't guaranty that props will be set:
     // https://github.com/facebook/react/issues/10389
-    if (this.videoRef.current) {
-      this.videoRef.current.muted = !!this.props.muted;
+    if (videoRef.current) {
+      videoRef.current.muted = !!muted;
     }
   };
 
-  renderComponent({ accessibility, ElementType, classes, unhandledProps }) {
-    const { controls, autoPlay, loop, poster, src } = this.props;
-
-    return (
-      <Ref innerRef={this.videoRef}>
-        <ElementType
-          autoPlay={autoPlay}
-          className={classes.root}
-          controls={controls}
-          loop={loop}
-          poster={poster}
-          src={src}
-          {...accessibility.attributes.root}
-          {...unhandledProps}
-        />
-      </Ref>
-    );
-  }
-}
+  React.useEffect(() => {
+    setVideoAttributes();
+  });
+
+  const { classes, styles: ResolvedStyles } = useStyles(Video.displayName, {
+    className: videoClassName,
+    mapPropsToStyles: () => ({
+      variables,
```

Comment:
Variables and others should be under `mapPropsToInlineStyles`

Reply:
Done